.github/workflows/build-gem.yml

```
@@ -0,0 +1,138 @@
+name: Build gem
```

Minor: Perhaps 

```suggestion
name: Package experimental gem
```

to make it a bit more clear what we're doing here?

---

An improvement I wanted to make was building both the alpha gem and the final gem.

This way when doing an actual release:

- we do not have to build the gem locally
- we are confident that the artifact is produced from a clean environment, and fully tested (more tests than `gem install` should probably be added)
- we can download it straight from the artifact, and eventually even have it added automatically (by that I mean something like a manual workflow instead of download+drag+drop, not necessarily _entirely_ automatically) to the GitHub release entry.

---

This is interesting! For that we'd need to tweak the version numbering logic, but otherwise seems like this PR would leave us in a good shape :+1: